### PR TITLE
Add spl-feature-proposal-cli and documentation

### PR DESCRIPTION
**Pull request by mvines**
```
$ spl-feature-proposal --help
spl-feature-proposal-cli 1.0.0
SPL Feature Proposal Command-line Utility

USAGE:
    spl-feature-proposal [FLAGS] [OPTIONS] <SUBCOMMAND>

FLAGS:
    -h, --help       Prints help information
    -V, --version    Prints version information
    -v, --verbose    Show additional information

OPTIONS:
    -C, --config <PATH>        Configuration file to use [default: /Users/mvines/.config/solana/cli/config.yml]
        --url <URL>            JSON RPC URL for the cluster [default: value from configuration file]
        --keypair <KEYPAIR>    Filepath or URL to a keypair [default: client keypair]

SUBCOMMANDS:
    address    Display address information for the feature proposal
    help       Prints this message or the help of the given subcommand(s)
    propose    Initiate a feature proposal
    tally      Tally the current results for a proposed feature
```

### CODE REVIEW

**Comment by mvines** [comment]
hey @CriesofCarrots / @t-nelson: this is all working well and I'd love some review of the overall concept as well is the code.  

@CriesofCarrots, `spl-feature-proposal-cli address ...` is what you'd use to generate the feature ids for the new inflation activation features that will be voted on by the community

File: docs/src/feature-proposal.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+After the feature is implemented and deployed to the Solana cluster,
+the *feature id* will be visible in `solana feature status` and the *feature
+proposer* may initiate the community proposal process.
+
+This is done by running:
+```
+$ spl-feature-proposal propose feature-proposal.json
+Feature Id: HQ3baDfNU7WKCyWvtMYZmi51YPs7vhSiLn1ESYp3jhiA
+Token Mint Address: ALvA7Lv9jbo8JFhxqnRpjWWuR3aD12uCb5KBJst4uc3d
+Delivery Token Address: GK55hNft4TGc3Hg4KzbjEmju8VfaNuXK8jQNDTZKcsNF
```
[the remark this answers is not in the document]
Comment (CriesofCarrots) [reply docs/src/feature-proposal.md:102]: I know you use `delivery_token_account` and `delivery_token_address` throughout the program. If you can stand the churn, I think something more like `distributor_token_account` would be easier to understand across the board. If not, I'd settle for just changing this print.

**Review by CriesofCarrots — COMMENTED** [review]
I did a slow-scroll review of the complete program code and tests in addition to this PR, and generally looks awesome to me. No qualms about the concepts.

File: feature-proposal/cli/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,452 @@
...
+                        .help("The address of the feature proposal"),
+                ),
+        )
+        .subcommand(
+            SubCommand::with_name("propose")
+                .about("Initiate a feature proposal")
+                .arg(
+                    Arg::with_name("feature_proposal")
+                        .value_name("FEATURE_PROPOSAL_KEYPAIR")
+                        .validator(is_keypair)
```
[the remark this answers is not in the document]
Comment (mvines) [reply feature-proposal/cli/src/main.rs:91]: Every time I try to use `Signer` trait and `DefaultSigner` stuff, I run into weird errors and everything falls apart.  I can barely follow how that works in spl-token cli.  Maybe we need some better documentation around how to hold it.  It's too hard and I just give up.

File: docs/src/feature-proposal.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (mvines) [reply docs/src/feature-proposal.md:102]: `distributor_token_account` sgtm, I'll make that change

[the remark this answers is not in the document]
Comment (mvines) [reply docs/src/feature-proposal.md]: dun

File: feature-proposal/cli/src/main.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (CriesofCarrots) [reply feature-proposal/cli/src/main.rs:91]: 🤕  Later, then. I'll add an issue about improving the usability of the `Signer` trait.

[the remark this answers is not in the document]
Comment (mvines) [reply feature-proposal/cli/src/main.rs:91]: Awesome.  I even tried to use `Signer` in the simple cli template I put together.  I beat my head for a while then just used `Keypair` and it worked: https://github.com/mvines/solana-cli-template/blob/146ad6206c69b87a3aeaaa2d9e941f6074810df0/src/main.rs#L16
I'm not proud of this though.

**Comment by mvines** [comment]
@t-nelson - I got merge happy but if I'll of course address any post-merge review comments you have

**Comment by t-nelson** [comment]
Sorry, I missed the review request this AM!

Looks great to me!  I agree with Tyera, that using `Signer` here would be nice, but also understand the ergonomics shortfall of its current state.  So I'm OK with punting for now

**Pull request merged**